sdk/src/wallet/update.rs

```diff
@@ -25,12 +26,34 @@ where
     crate::wallet::Error: From<S::Error>,
     crate::client::Error: From<S::Error>,
 {
-    /// Set the alias for the wallet.
-    pub async fn set_alias(&self, alias: &str) -> crate::wallet::Result<()> {
-        let mut wallet_data = self.data_mut().await;
-        wallet_data.alias = Some(alias.to_string());
+    /// Update the wallet address with a possible new Bech32 HRP and clear the inaccessible incoming transactions.
+    pub(crate) async fn update_address_hrp(&self) -> crate::wallet::Result<()> {
+        let bech32_hrp = self.client().get_bech32_hrp().await?;
+        log::debug!("updating wallet with new bech32 hrp: {}", bech32_hrp);
+
+        self.address_mut().await.hrp = bech32_hrp;
+
         #[cfg(feature = "storage")]
-        self.storage_manager().save_wallet_data(&wallet_data).await?;
+        {
+            let wallet_ledger = {
+                let mut wallet_ledger = self.ledger_mut().await;
+                wallet_ledger.inaccessible_incoming_transactions.clear();
+                WalletLedgerDto::from(&*wallet_ledger)
+            };
+
+            log::debug!("[save] wallet ledger with updated bech32 hrp",);
+            self.storage_manager().save_wallet_ledger(&wallet_ledger).await?;
+        }
+
+        Ok(())
+    }
+
+    /// Update the alias for the wallet.
+    pub async fn update_wallet_alias(&self, alias: &str) -> crate::wallet::Result<()> {
```

Comment:
`update_alias`

Reply:
I know you probably changed set to update to match `update_address_hrp` but I personally prefer set. I'd rename everything to set.

Reply:
this is done